### ISSUE TITLE
Should be able to manually enter duration

### ISSUE THREAD

**Issue by cfilipov** [cfilipov/MuscleBook#8]
When tapping the "duration" row in the "Add data point" screen, an option to manually specify duration should be provided. An alert view with an input field would be sufficient. The start time row would also need to be re-enabled to allow changes.

**Comment by JonathanGuberman** [comment]
Related (not sure if it's worth making another issue for it), it would be great to be able to manually change the date as well. When I first loaded the app I immediately wanted to enter my last few workouts into it to get an idea for how it works, and, as far as I can tell, there's no way to do that.

**Comment by cfilipov** [comment]
I thought I added an issue for this, but I guess I forgot. The start time and start date were originally editable before I added the stopwatch. I disabled them because it simplified the logic for handling and invalidating the stopwatch. I immediately regretted this decision when I started logging workouts.

Also worth pointing out (and I should add this to the issues as well, it's currently only mentioned in the changelog's known issues). It's not currently possible to enter past workouts if you already have workout data (due to the fact that time-bounds checking needs to be done and isn't implemented).

**Comment by cfedk** [comment]
Tangentially related: Should duration be a required field? While it's good to have for running, biking and perhaps rowing, I wouldn't personally use that for weight exercises. Start-stopping the timer feels... dirty...

**Comment by JonathanGuberman** [comment]
I agree that it probably should be _required_, but I think it could actually be really useful: it can let you see how difficult a set is and track progress over time ("last week I really struggled through that 5x250 set, it took me 45 seconds, but today I blew through it in 25 seconds!").

I also agree that it's not the workflow I'm used to—I tend to go home and log everything after the fact—but it appears as if this app is meant to be used while you're doing the sets, to keep track of all this timing data (which implicitly includes how much rest you're taking between each set). That appeals to me, because it gives you more data to work with later on! I'm going to try using it as intended when I go to the gym tomorrow. 😃

**Comment by cfedk** [comment]
By no means saying it should be removed! It is clearly there because @cfilipov finds it useful!

That has also always been my approach - unfortunately my current gym's rules prevent me from testing whether this app would change my mind.

**Comment by cfilipov** [comment]
As @JonathanGuberman said, it can be useful to know tempo, and it also allows me to calculate total active time and time between sets. That said, I did intend to make duration optional (where it's not required, exercises like planks would require it) because not every lifter cares about tempo or time between sets. I did intend to keep start date and start time as required because it gives a logical ordering at all sets.
